x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/openai/embeddings/OpenAiEmbeddingsServiceSettings.java

```diff
@@ -34,62 +37,108 @@
 import static org.elasticsearch.xpack.inference.services.ServiceUtils.removeAsType;
 
 /**
- * Defines the base settings for interacting with OpenAI.
+ * Defines the service settings for interacting with OpenAI's text embedding models.
  */
-public class OpenAiServiceSettings implements ServiceSettings {
+public class OpenAiEmbeddingsServiceSettings implements ServiceSettings {
 
     public static final String NAME = "openai_service_settings";
 
     public static final String ORGANIZATION = "organization_id";
+    static final String DIMENSIONS_SET_BY_USER = "dimensions_set_by_user";
 
-    public static OpenAiServiceSettings fromMap(Map<String, Object> map) {
+    public static OpenAiEmbeddingsServiceSettings fromMap(Map<String, Object> map, OpenAiParseContext context) {
+        return switch (context) {
+            case REQUEST -> fromRequestMap(map);
+            case PERSISTENT -> fromPersistentMap(map);
+        };
+    }
+
+    private static OpenAiEmbeddingsServiceSettings fromPersistentMap(Map<String, Object> map) {
+        ValidationException validationException = new ValidationException();
+
+        var commonFields = fromMap(map, validationException);
```

Comment:
The `dimensions` field should always be set when ready from the persistent settings. But since we're allowing null we'll leave it that way just in case. We'll check for null in a few places to handle it.